salt/utils/templates.py

```diff
@@ -59,9 +59,9 @@ def render_tmpl(tmplsrc,
         if 'salt' in kws:
             if 'cmd.run' in kws['salt'] and 'cmd.shell' in kws['salt']:
                 kws['salt']['cmd.run'] = kws['salt']['cmd.shell']
-            if 'run' in kws['salt'].get('cmd', {}) \
-                    and 'shell' in kws['salt'].get('cmd', {}):
-                kws['salt']['cmd']['run'] = kws['salt']['cmd']['shell']
+            if ('cmd' in kws['salt'] and hasattr(kws['salt']['cmd'], 'run')
```

Comment:
It does that [here](https://github.com/saltstack/salt/pull/22447/files#diff-96d7010eb773dc57ad2c8eb89cd12807L60) but this line is for the short notation that was added in 2014.7.